Date: Sat Mar 23 23:52:41 2024 +0900

    ttyd のサーバー側を突然終了させると、ttyd クライアントで
    再接続ループが発生する。これはそのワークアラウンド。

    どちらか片方を適用すればよい。
    どちらを適用しても同じような動きになるはず。

    参考: https://github.com/sockjs/sockjs-client/issues/176#issuecomment-45404353
          SockJS client not emitting close event when disconnected ungracefully #176

    * sockjs-client で対処する場合
	| SockJS (= WebSocket) の 接続クローズコードが 1000 以外の時は、
	| すべてエラー扱いにする (onError イベントを送出)
	|
	|
	|
	|diff --git a/node_modules/sockjs-client/lib/main.js b/node_modules/sockjs-client/lib/main.js
	|--- a/node_modules/sockjs-client/lib/main.js
	|+++ b/node_modules/sockjs-client/lib/main.js
	|@@ -351,7 +351,7 @@
	|   setTimeout(function() {
	|     this.readyState = SockJS.CLOSED;
	| 
	|-    if (forceFail) {
	|+    if (forceFail || (code || 1000) != 1000) {
	|       this.dispatchEvent(new Event('error'));
	|     }
	| 

    * ttyd クライアント側で対処する場合
	| 再接続のデフォルトを false に変更
	|
	| 接続後に false に変更できるオプション利用可能だが、これが意味
	| をなさなくなるので、接続後に true に変更するオプションを追加
	|
	|diff --git a/src/components/terminal/index.tsx b/src/components/terminal/index.tsx
	|--- a/src/components/terminal/index.tsx
	|+++ b/src/components/terminal/index.tsx
	|@@ -75,8 +75,8 @@
	|     private titleFixed: string;
	|     private resizeTimeout: number;
	|     private resizeOverlay = true;
	|-    private reconnect = true;
|-    private doReconnect = true;
	|+    private reconnect = false;
	|+    private doReconnect = false;
	| 
	|     constructor(props: Props) {
	|         super(props);
	|@@ -307,6 +307,13 @@
	|                         this.doReconnect = false;
	|                     }
	|                     break;
	|+                case 'enableReconnect':
	|+                    if (value) {
	|+                        console.log(`[ttyd] Reconnect enabled`);
	|+                        this.reconnect = true;
	|+                        this.doReconnect = true;
	|+                    }
	|+                    break;
	|                 case 'titleFixed':
	|                     if (!value || value === '') return;
	|                     console.log(`[ttyd] setting fixed title: ${value}`);

diff --git a/src/components/terminal/xterm/index.ts b/src/components/terminal/xterm/index.ts
--- a/src/components/terminal/xterm/index.ts
+++ b/src/components/terminal/xterm/index.ts
@@ -96,8 +96,8 @@
     private title?: string;
     private titleFixed?: string;
     private resizeOverlay = true;
-    private reconnect = true;
-    private doReconnect = true;
+    private reconnect = false;
+    private doReconnect = false;
 
     private writeFunc = (data: ArrayBuffer) => this.writeData(new Uint8Array(data));
 
@@ -402,6 +402,13 @@
                         this.doReconnect = false;
                     }
                     break;
+                case 'enableReconnect':
+                    if (value) {
+                        console.log(`[ttyd] Reconnect enabled`);
+                        this.reconnect = true;
+                        this.doReconnect = true;
+                    }
+                    break;
                 case 'enableZmodem':
                     if (value) console.log('[ttyd] Zmodem enabled');
                     break;
